parser/testdata/query/format/query_with_expr_compare.sql

```diff
@@ -0,0 +1,10 @@
+-- Origin SQL:
+SELECT date, path, splitByChar('/', path)[2] AS path_b
+FROM(
+    SELECT 'pathA/pathB/pathC' AS path, '2024-09-10' AS date
+    )
+WHERE toDate(date) = '2024-09-01' AND '2024-09-30'
```

Comment:
```suggestion
WHERE toDate(date) between '2024-09-01' AND '2024-09-30'
```

I don't really understand the meaning of this SQL, so I try to run it in the clickhouse client and it gets an error. 
So it's really non-grammatical? Maybe '=' should be 'between'?

Reply:
<img width="1079" alt="image" src="https://github.com/user-attachments/assets/d375b7f0-4544-4438-be01-8b2b4b41cd90">

Reply:
@Lance726 Yes, you're right. Should be BETWEEN instead of `=`.